ethpm/package.py

```diff
@@ -11,27 +12,69 @@
 
 
 class Package(object):
-    def __init__(self, lockfile):
+    web3 = None
+
+    def __init__(self, package_identifier, web3=None):
         """
         A lockfile can be:
         - filesystem path
         - parsed lockfile JSON
         - lockfile URI
         """
-        if not os.path.exists(lockfile):
+        self.web3 = web3
+
+        if not os.path.exists(package_identifier):
             raise ValidationError
 
-        self.package_identifier = lockfile
+        self.package_identifier = package_identifier
 
-        schema_data = json.load(open('./ethpm/lockfileSpecification.json'))
-        package_data = json.load(open(lockfile))
+        schema_data = self._load_json('./ethpm/assets/release-lockfile.schema.v1.json')
+        package_data = self._load_json(package_identifier)
 
         try:
             validate(package_data, schema_data)
         except jsonValidationError:
             raise ValidationError
         self.package_data = package_data
 
+    def _load_json(self, path):
+        return json.load(open(path))
+
+    def set_default_web3(self, web3):
+        """
+        Set the default Web3 instance.
+        """
+        self.web3 = web3
+
+    def get_contract_type(self, name, web3=None):
+        """
+        API to generate a contract factory class.
+        """
+        self._validate_contract_name(name)
+        current_web3 = None
+
+        if web3 is not None:
+            current_web3 = web3
+        else:
+            current_web3 = self.web3
+        if current_web3 is None:
+            raise ValidationError
+
+        if name in self.package_data['contract_types']:
+            contract = self.package_data['contract_types'][name]
+            contract_factory = current_web3.eth.contract(
+                abi=contract["abi"],
```

Comment:
the `abi` key is not guaranteed to be present.  Neither are `bytecode` or `runtime_bytecode`.  would recommend splitting this logic up as follows.

* a function `validate_minimal_contract_data_present(contract_data)` which validates that at least one of the `abi/bytecode/runtime_bytecode` keys are present.
* a function `generate_contract_factory_kwargs` which uses the `eth_utils.to_dict` decorator to build a dictionary of the keyword arguments to be passed to `web3.eth.contract`

Then you can do this:

```python
v

```python
validate_minimal_contract_data_present(contract_data)
contract_kwargs = generate_contract_factory_kwargs(contract_data)
contract_factory = w3.eth.contract(**contract_kwargs)
```